src/dd/RealNumber.cpp

```diff
@@ -79,32 +105,6 @@ bool RealNumber::approximatelyZero(const RealNumber* e) noexcept {
   return e == &constants::zero || approximatelyZero(val(e));
 }
 
-bool RealNumber::noRefCountingNeeded(const RealNumber* const num) noexcept {
-  assert(!isNegativePointer(num));
-  return num == nullptr || constants::isStaticNumber(num) ||
-         num->ref == std::numeric_limits<RefCount>::max();
-}
```

Comment:
Would it make sense to even preserve this function? and use it in the corresponding `mark()` function? Just thinking about making the diff of this PR as small as possible.